models/groups.go

```diff
@@ -148,39 +149,62 @@ func (g *Group) BuildSuccessfulJsonResponse(ctx context.Context) ([]byte, error)
 	return jsonResponse, nil
 }
 
-type CreateGroup struct {
+type CreateUpdateGroup struct {
 	Description *string `json:"name"`
 	Precedence  *int64  `json:"precedence"`
-	GroupName   string
+	GroupName   *string
+	GroupsList  *cognitoidentityprovider.ListGroupsOutput
 }
 
-func (g *CreateGroup) ValidateCreateGroupRequest(ctx context.Context) []error {
+//ValidateCreateGroupRequest validate the create group request
+func (g *CreateUpdateGroup) ValidateCreateUpdateGroupRequest(ctx context.Context) []error {
 	var validationErrs []error
 
 	if g.Description == nil {
 		validationErrs = append(validationErrs, NewValidationError(ctx, InvalidGroupName, MissingGroupName))
 	} else if m, _ := regexp.MatchString("(?i)^role_.*", *g.Description); m {
 		validationErrs = append(validationErrs, NewValidationError(ctx, InvalidGroupName, IncorrectPatternInGroupName))
+	} else {
+		//ensure groups name in description doesn't already exist - creation only
+		if g.GroupsList != nil {
+			g.CleanGroupDescription()
```

Comment:
Are we still cleaning the Group description?
I thought it was meant to be human readable now, as we have the UUID for urls etc.